docs/tutorial.md

```diff
@@ -1,5 +1,910 @@
+them in their own function:
+
+```js
+const StoryThumbnail = props => h(
+  "li",
+  {class: {
+      unread: props.unread,
+      reading: props.reading,
+  }},
+  [
+    h("p", {class: "title"}, emphasize(props.filter, props.title)),
+    h("p", {class: "author"}, props.author)
+  ]
+)
+```
+
+> The last example demonstrates a helpful feature of the `class` property. When
+> you set it to an object rather than a string, each key with a truthy value
+> will become a class in the class list.
+
+Continue by creating functions for each section of the view:
+
+```js
+
+const StoryList = props => h("div", {class: "stories"}, [
+  h("ul", {}, Object.keys(props.stories).map(id =>
+    StoryThumbnail({
+      id,
+      title: props.stories[id].title,
+      author: props.stories[id].author,
+      unread: !props.stories[id].seen,
+      reading: props.reading === id,
+      filter: props.filter,
+    })
+  ))
+])
+
+const Filter = props => h("div", {class: "filter"}, [
+  "Filter:",
+  h("span", {class: "filter-word"}, props.filter),
+  h("button", {}, "\u270E")
+])
+
+const StoryDetail = props => h("div", {class: "story"}, [
+  props && h("h1", {}, props.title),
+  props && h("p", {}, `
+    Lorem ipsum dolor sit amet, consectetur adipiscing
+    elit, sed do eiusmod tempor incididunt ut labore et
+    dolore magna aliqua. Ut enim ad minim veniam, qui
+    nostrud exercitation ullamco laboris nisi ut aliquip
+    ex ea commodo consequat.
+  `),
+   props && h("p", {class: "signature"}, props.author)
+])
+
+const AutoUpdate = props => h("div", {class: "autoupdate"}, [
+  "Auto update: ",
+  h("input", {type: "checkbox"})
+])
+
+const Container = content => h("div", {class: "container"}, content)
+
+```
+
+
+With those, the view can be written as:
+
+```js
+view: () => Container([
+  Filter({
+    filter: "ocean"
+  }),
+  StoryList({
+    stories: {
+      "112": {
+        title: "The Ocean is Sinking",
+        author: "Kat Stropher",
+        seen: false,
+      },
+      "113": {
+        title: "Ocean life is brutal",
+        author: "Surphy McBrah",
+        seen: true,
+      },
+      "114": {
+        title: "Family friendly fun at the ocean exhibit",
+        author: "Guy Prosales",
+        seen: true,
+      }
+    },
+    reading: "113",
+    filter: "ocean"
+  }),
+  StoryDetail({
+    title: "Ocean life is brutal",
+    author: "Surphy McBrah",
+  }),
+  AutoUpdate(),
+])
+```
+
+What you see on the page should be exactly the same as before, because we haven't
+changed what `view` returns. Using basic functional composition, we were able to make
+the code a bit more manageable, and that's the only difference.
+
+State
+-------------------------------
+
+With all that view logic broken out in separate functions, `view` is starting to look like
+plain _data_. The next step is to fully separate data from the view.
+
+Add an `init` property to your app, with this pure data:
+
+```js
+  init: {
```

Comment:
I'm curious if you want to change this to `init: () => ({ ...stateStuffHere })` so it's closer to an Action. I think it's fine to use a plain object, but I think we've discussed that behaviour before as a "happy accident," so I'm not certain we want to advertise it.

Reply:
I think the pay-off here will be when a user wants to fire an effect immediately, that will be easier to understand if `init` is an action, rather than describing this one-off direct call to dispatch.

Reply:
The downside is actions are the next section.

Reply:
Yeah and also, if I understand @jorgebucaran correctly, he doesn't want us to think of `init` as an action even though it works like one. (I could be wrong)

Reply:
Yeah, that's fair.